### PR TITLE
763 rmc 115 generate initial

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
Single print file formats for England Wales and NI

# What has changed
Tests to check for this

# How to test?
Download related actionscheduler & actionexporter.
mvn clean install them both locally, make up docker dev and fire off the tests

# Links
https://trello.com/c/ZmGZEf9A/763-rmc-115-action-exporter-to-generate-initial-contact-print-files-13
# Screenshots (if appropriate):

### CODE REVIEW

File: acceptance_tests/features/sample_file_smoke.feature [hunk 1]
```diff
@@ -15,10 +14,30 @@ Feature: Smoke tests to check the system is tied together correctly
     And correctly formatted "P_IC_ICL2" print files are created
     And there is a correct "P_IC_ICL2" manifest file for each csv file written
 
+  Scenario: Successful sample file upload and NI ICL print file
+    Given an action rule of type ICL4E is set 10 seconds in the future
```
Comment (NickGrantX) [inline acceptance_tests/features/sample_file_smoke.feature]: This should be type `ICL4N`.

E = England
W = Wales
N = Northern [Ireland]

File: acceptance_tests/features/steps/action_steps.py [hunk 2]
```diff
@@ -15,25 +15,16 @@ def setup_action_rule(context, action_type, action_rule_delay):
     trigger_date_time = (datetime.utcnow() + timedelta(seconds=int(action_rule_delay))).isoformat() + 'Z'
 
     classifiers_for_action_type = {
-        'ICL1E': {'treatmentCode': ['HH_LF3R2E', 'HH_LF3R3AE', 'HH_LF3R3BE', 'HH_LFNR1E', 'HH_LF2R3BE']},
+        'ICL1E': {'treatmentCode': ['HH_LFNR1E', 'HH_LFNR2E', 'HH_LFNR3AE', 'HH_LF2R1E', 'HH_LF2R2E', 'HH_LF2R3AE',
+                                    'HH_LF2R3BE', 'HH_LF3R1E', 'HH_LF3R2E', 'HH_LF3R3AE', 'HH_LF3R3BE']},
         'ICL2W': {'treatmentCode': ['HH_LFNR1W', 'HH_LFNR2W', 'HH_LFNR3AW', 'HH_LF2R1W', 'HH_LF2R2W', 'HH_LF2R3AW',
                                     'HH_LF2R3BW', 'HH_LF3R1W', 'HH_LF3R2W', 'HH_LF3R3AW', 'HH_LF3R3BW']},
+        'ICL4E': {'treatmentCode': ['HH_1LSFN', 'HH_2LEFN']},
```
Comment (NickGrantX) [inline acceptance_tests/features/steps/action_steps.py]: change to `ICL4N`

File: acceptance_tests/utilities/print_file_helper.py [hunk 3]
```diff
@@ -27,12 +27,13 @@ def create_expected_csv_lines(context, prefix):
...
 
-def create_expected_wales_csv_lines(context, prefix):
+def create_expected_questionaire_csv_lines(context, prefix):
     actual_data = defaultdict(dict)
 
+    #
     for message in context.messages_received:
         if message['event']['type'] == 'UAC_UPDATED':
-            if message['payload']['uac']['questionnaireId'][:2] == '02':
+            if message['payload']['uac']['questionnaireId'][:2] in ('01', '02', '04'):
```
Comment (NickGrantX) [inline acceptance_tests/utilities/print_file_helper.py]: we are potentially allowing a regression bug here, where multiple UAC/QID pairs get issued for anything other than a Welsh initial contact questionnaire

File: resources/sample_files/sample_input_england_census_spec.csv [hunk 4]
```diff
@@ -1,6 +1,26 @@
...
+DDR190314000000195675,DDR190314000000113740,10008677190,HH,Household,U,RD06,,Flat 56 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_LF2R3BE,3
+DDR190314000000195675,DDR190314000000113740,10008677190,HH,Household,U,RD06,,Flat 57 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_LF3R1E,3
+DDR190314000000195675,DDR190314000000113740,10008677190,HH,Household,U,RD06,,Flat 58 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_LF3R2E,3
+DDR190314000000195675,DDR190314000000113740,10008677190,HH,Household,U,RD06,,Flat 59 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_LF3R3AE,3
+DDR190314000000195675,DDR190314000000113740,10008677190,HH,Household,U,RD06,,Flat 50 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_LF3R3BE,3
+
```
Comment (NickGrantX) [inline resources/sample_files/sample_input_england_census_spec.csv]: why all these blank lines

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Action type should be `ICL4N` not `ICL4E` for Northern Ireland

File: acceptance_tests/features/steps/print_file_steps.py [hunk 5]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
     )
 
     return manifest
+
+
+def _get_country_and_purpose(prefix):
+    if "P_IC_ICL1" in prefix:
+        country = 'England'
+        purpose = 'Initial contact letter households'
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]: Unless these variables are supposed to change with further checks to the prefix, we can probably return the tuple early and skip assignment altogether.

```suggestion
        return 'Initial contact letter households', 'England'
```

Also, is `in` the right thing to use for the check here?

File: acceptance_tests/features/steps/print_file_steps.py [hunk 6]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
...
+
+    if "P_IC_ICL2" in prefix:
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]:
```suggestion
    elif "P_IC_ICL2" in prefix:
```

File: acceptance_tests/features/steps/print_file_steps.py [hunk 7]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
...
+        country = 'Wales'
+        purpose = 'Initial contact letter households'
+
+    if "P_IC_ICL4" in prefix:
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]:
```suggestion
    elif "P_IC_ICL4" in prefix:
```

File: acceptance_tests/features/steps/print_file_steps.py [hunk 8]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
...
+        country = 'Northern Ireland'
+        purpose = 'Initial contact letter households'
+
+    if "P_IC_H1" in prefix:
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]:
```suggestion
    elif "P_IC_H1" in prefix:
```

File: acceptance_tests/features/steps/print_file_steps.py [hunk 9]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
...
+        country = "England"
+        purpose = 'Initial contact questionnaire households'
+
+    if "P_IC_H2" in prefix:
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]:
```suggestion
    elif "P_IC_H2" in prefix:
```

File: acceptance_tests/features/steps/print_file_steps.py [hunk 10]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
...
+        country = 'Wales'
+        purpose = 'Initial contact questionnaire households'
+
+    if "P_IC_H4" in prefix:
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]:
```suggestion
    elif "P_IC_H4" in prefix:
```

File: acceptance_tests/features/steps/print_file_steps.py [hunk 11]
```diff
@@ -154,3 +159,31 @@ def _create_expected_manifest(sftp_utility, csv_file, created_datetime, prefix):
...
+        country = "Northern Ireland"
+        purpose = 'Initial contact questionnaire households'
+
+    return purpose, country
```
Comment (jcox-dev) [inline acceptance_tests/features/steps/print_file_steps.py]: Do we need a default (e.g.: `None, None`)?

File: resources/sample_files/sample_input_census_spec_england_questionnaire.csv [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+DDR190314000000195675,DDR190314000000113741,,HH,Household,U,RD06,,Flat 566 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_QF2R2E,3
+DDR190314000000195675,DDR190314000000113742,,HH,Household,U,RD06,,Flat 561 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_QF2R3AE,3
+DDR190314000000195675,DDR190314000000113743,,HH,Household,U,RD06,,Flat 562 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_QF3R1E,3
+DDR190314000000195675,DDR190314000000113744,,HH,Household,U,RD06,,Flat 563 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_QF3R2E,3
+DDR190314000000195675,DDR190314000000113745,,HH,Household,U,RD06,,Flat 564 Francombe House,Commercial Road,,Windleybury,XX1 0XX,51.4463421,-2.5924477,E00073438,E01014540,E02003043,E06000023,E12000009,1,5,1,2,HH_QF3R3AE,3
+
```
Comment (jcox-dev) [inline resources/sample_files/sample_input_census_spec_england_questionnaire.csv]: Empty lines?

File: resources/sample_files/sample_input_england_census_spec.csv (hunk 4 above)
Comment (LukeLoze1) [reply resources/sample_files/sample_input_england_census_spec.csv]: The space is symbolic of

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Tests pass (is this a false-positive, following Nick's comment?) locally just some utility function comments.

File: acceptance_tests/features/steps/print_file_steps.py (hunk 11 above)
Comment (LukeLoze1) [reply acceptance_tests/features/steps/print_file_steps.py]: I've decided to assertFalse instead to catch this early

**Review by jcox-dev — APPROVED** [review]

**Review by NickGrantX — APPROVED** [review]
LTGM 👍

**Review by hbrace — APPROVED** [review]

**Pull request merged**